web/src/reload.js

```diff
@@ -40,9 +40,7 @@ function preserve_state(send_after_reload, save_pointer, save_narrow, save_compo
         return;
     }
 
-    if (!message_lists.home) {
-        // If we haven't yet initialized the message_lists module,
-        // we can't preserve state across reloads.
+    if (!$("#app-loading").hasClass("loaded")) {
```

Comment:
Hmm, I'm not sure what's correct here. In theory, we could do the `save_compose` logic even if the message lists are not in a good state, and push a `message_lists.home` check a bit deeper, below.

At the same time, the `message_lists.current` logic probably doesn't make sense if we don't actually have a proper message ID selected yet; reading `narrow.js`, there's likely some races there.

I think we might want to try skipping this commit and doing an independent little project to rewrite this function using proper data structures written into local storage rather than the legacy URL string model, `zod` for parsing, and a more robust set of checks for which of these bonus things to include, and some more thought-through algorithm for when we should give up or maybe try to use the values from the last reload for the narrow position logic, because this app load failed before getting that metadata pushed down into the actual web app's state.

Reply:
I guess this is probably unlikely to make things worse, so maybe it's fine to merge this commit rather than skipping while still keeping in mind this module has flawed logic for various error situations.

Reply:
Yeah, I have a lot of changes in the pipeline here after these commits and it would help to have this merged, we can thing abour refactoring when the dust is settled on how we want to eliminate populating message_lists.home initially.